tools/sim/test/test_carla_integration.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+import sys
+import time
+import unittest
+
+from cereal import messaging
+
+from common.params import Params
+from selfdrive.manager import manager
+from tools.sim import bridge
+from tools.sim.bridge import connect_carla_client
+
+
+class TestCarlaIntegration(unittest.TestCase):
+  """
+  Tests need Carla simulator to run
+  """
+  subprocess = None
+
+  def test_connect_with_carla(self):
+    # Test connecting to Carla within 5 seconds and return no RuntimeError
+    client = connect_carla_client()
+    assert client is not None
+    # Will raise an error if not connected
+    client.load_world('Town04_Opt')
+
+  def test_run_bridge(self):
+    # Test bridge connect with carla and runs without any errors for 60 seconds
+    test_duration = 60
+
+    params = Params()
+    args = sys.argv[2:]  # Remove test arguments when executing this test
+    args.append('--low_quality')
```

Comment:
why are the args from the test being copied? works now because it's just an empty list

Reply:
This was to execute it in command line, and would fail otherwise. But I will just remove it. 
python -m unittest test_carla_integration.TestCarlaIntegration.test_engage